src/UCommerce.Transactions.Payments.Adyen/AdyenPaymentMethodService.cs

```diff
@@ -50,17 +51,23 @@ public override Payment Extract(HttpRequest httpRequest)
             var jsonObj = (JObject?)JsonConvert.DeserializeObject(contentJson);
             var reference = jsonObj?["notificationItems"]?[0]?["NotificationRequestItem"]?[PaymentReferenceKey]
                 ?.Value<string>();
+            var eventCode = jsonObj?["notificationItems"]?[0]?["NotificationRequestItem"]?[EventCodeKey]
+    ?.Value<string>();
 
-            return _paymentRepository.Select(x => x.ReferenceId == reference).FirstOrDefault() ??
-                   throw new NullReferenceException(
-                       $"Could not find a payment with ReferenceId: '{reference}'.");
+            if(string.IsNullOrWhiteSpace(reference) && eventCode == "REPORT_AVAILABLE")
+            {
+                _loggingService.Information<AdyenPaymentMethodService>("We received a report webhook");
+                HttpContext.Current.Response.Write("[accepted]");
+                HttpContext.Current.Response.End();

```

Comment:
These to lines are duplicated, means used two places, action move them to a method ;)

Something like

protected virtuel void CompleteHttpRequest() {
                HttpContext.Current?.Response.Write("[accepted]");
                HttpContext.Current?.Response.End();
}